StripeCore/StripeCore/Source/DownloadManager/DownloadManager.swift

```diff
@@ -9,37 +9,25 @@ import UIKit
 
 /// For internal SDK use only.
 @objc(STP_Internal_DownloadManager)
-@_spi(STP) public class DownloadManager: NSObject, URLSessionDelegate {
+@_spi(STP) public class DownloadManager: NSObject {
     public typealias UpdateImageHandler = (UIImage) -> Void
 
-    private enum Error: Swift.Error {
-        case downloadSyncFailure
-        case downloadAsyncFailure
+    enum Error: Swift.Error {
+        case failedToMakeImageFromData
     }
 
     public static let sharedManager = DownloadManager()
 
-    let downloadQueue: DispatchQueue
-    let downloadOperationQueue: OperationQueue
-    let session: URLSession!
+    let imageCache = ImageCache()
 
-    var imageCache: [String: UIImage]
-    var pendingRequests: [String: URLSessionTask]
-    var updateHandlers: [String: [UpdateImageHandler]]
-
-    let imageCacheSemaphore: DispatchSemaphore
-    let pendingRequestsSemaphore: DispatchSemaphore
-
-    let STPCacheExpirationInterval = (60 * 60 * 24 * 7)  // 1 week
-    var urlCache: URLCache?
+    private let session: URLSession
+    private let analyticsClient: STPAnalyticsClient
+    private var diskCache: URLCache?
 
     public init(
-        urlSessionConfiguration: URLSessionConfiguration = .default
+        urlSessionConfiguration: URLSessionConfiguration = .default,
+        analyticsClient: STPAnalyticsClient = .sharedClient
     ) {
-        downloadQueue = DispatchQueue(label: "Stripe Download Cache", attributes: .concurrent)
-        downloadOperationQueue = OperationQueue()
-        downloadOperationQueue.underlyingQueue = downloadQueue
-
         let configuration = urlSessionConfiguration
         if let cachesURL = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)
```

Comment:
Why create a custom URL instead of using the default?

Reply:
I don't see a great reason why we need a custom URL here, but given that is what we already have I think it makes sense to keep it.
1. I can't see any drawbacks to using a custom URL
2. If we change it and a merchant updates the SDK they'd have this existing cache directory, but then we'd be putting stuff in the default URL which seems odd.
3. Merchant loses existing cache on SDK update

Reply:
Maybe one drawback is that creating the URL could fail, in which case we'd silently lose caching.  I don't have a sense for how possible that is. I'm fine to keep since it's existing behavior - maybe it makes sense to have it separate from the default cache so we can delete it if we want.